python/ray/workflow/step_executor.py

```diff
@@ -355,8 +429,15 @@ def _workflow_step_executor(
                 "is not allowed.")
         assert not isinstance(persisted_output, Workflow)
     else:
+        # validate the checkpoint options before commit the step
+        _validate_checkpoint_options(persisted_output)
```

Comment:
It looks bad since it's in the runtime instead of compiling time. Should we just print warning so that it won't fail the workflow?